### PR TITLE
[Infra Views Read/Write]  Create Databag item

### PR DESCRIPTION
**Pull request by chaitali-mane**
### :nut_and_bolt: Description: What code changed, and why?
Need to add Create Data Bag Items UI. 
auto-create-databag-item1

auto-create-databag-item2

### :chains: Related Resources
[4730](https://github.com/chef/automate/issues/4730)

### :+1: Definition of Done
I have added UI changes to create the data bag item.

### :athletic_shoe: How to Build and Test the Change
Go to the `infrastructure tab` from the top navigation bar then you can see the `Chef Server`
If you have data then you can see the list of servers --> then click to any of the server --> see the list of orgs 
--> click to any of org ---> you can see the multiple tabs
--> click to `Databags` tab
--> see the list of Databags ---> click to any of databag --> you are on databag details page 
--> click to Create Item button --> modal will be open
--> Fill data, ID field is mandatory 

### :white_check_mark: Checklist

- [ ] I have read the [CONTRIBUTING document](https://github.com/chef/automate/blob/master/CONTRIBUTING.md).
- [ ] Tests added/updated?
- [ ] Docs added/updated?
- [ ] All commits have been signed-off for the [Developer Certification of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### :camera: Screenshots, if applicable

https://user-images.githubusercontent.com/71449322/111597042-d1b47780-87f3-11eb-8b21-a4833c7bf5f8.mp4

### CODE REVIEW

File: components/automate-ui/src/app/modules/infra-proxy/data-bags-details/data-bags-details.component.html [hunk 1]
```diff
@@ -26,9 +26,20 @@
           <chef-option value='details' data-cy="details-tab">Details</chef-option>
         </chef-tab-selector>
       </chef-page-header>
+
+      <app-create-databag-item-modal
+      [openEvent]="openDataBagItemModal"
+      [server_Id]="serverId"
+      [org_Id]="orgId"
+      [name]="dataBagsName"
+      [currentPage]="current_page"></app-create-databag-item-modal>
```
Comment (vivekshankar1) [inline components/automate-ui/src/app/modules/infra-proxy/data-bags-details/data-bags-details.component.html:35]:
```suggestion
      <app-create-databag-item-modal
          [openEvent]="openDataBagItemModal"
          [server_Id]="serverId"
          [org_Id]="orgId"
          [name]="dataBagsName"
          [currentPage]="current_page">
      </app-create-databag-item-modal>
```

File: components/automate-ui/src/app/modules/infra-proxy/create-databag-item-modal/create-databag-item-modal.component.ts [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+import { EntityStatus, pending } from 'app/entities/entities';
+import { HttpStatus } from 'app/types/types';
+import {
+  saveStatus,
+  saveError
+} from 'app/entities/data-bags/data-bag-details.selector';
+import { DataBagItem } from 'app/entities/data-bags/data-bags.model';
+import {
+  CreateDataBagItem,
+  GetDataBagItems } from 'app/entities/data-bags/data-bag-details.actions';
```
Comment (vivekshankar1) [inline components/automate-ui/src/app/modules/infra-proxy/create-databag-item-modal/create-databag-item-modal.component.ts:25]:
```suggestion
import {
  CreateDataBagItem,
  GetDataBagItems 
} from 'app/entities/data-bags/data-bag-details.actions';
```

File: components/automate-ui/src/app/modules/infra-proxy/create-databag-item-modal/create-databag-item-modal.component.spec.ts [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(CreateDatabagItemModalComponent);
+    component = fixture.componentInstance;
+    component.openEvent = new EventEmitter();
+    fixture.detectChanges();
+  });
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
```
Comment (vivekshankar1) [inline components/automate-ui/src/app/modules/infra-proxy/create-databag-item-modal/create-databag-item-modal.component.spec.ts:55]: Could we add actual test here?

File: components/automate-ui/src/app/entities/data-bags/data-bag-details.effects.ts [hunk 4]
```diff
@@ -43,4 +48,33 @@ export class DataBagItemsEffects {
...
+    mergeMap(({ payload: { dataBagItem } }: CreateDataBagItem) =>
+      this.requests.createDataBagItem(dataBagItem).pipe(
+        map((resp: CreateDataBagItemSuccessPayload) => new CreateDataBagItemSuccess(resp)),
+        catchError((error: HttpErrorResponse) =>
+          observableOf(new CreateDataBagItemFailure(error))))));
+
+  @Effect()
+  createDataBagItemSuccess$ = this.actions$.pipe(
+    ofType(DataBagItemsActionTypes.CREATE_SUCCESS),
+    map(({ payload: { id : id } }: CreateDataBagItemSuccess) => {
```
Comment (vivekshankar1) [inline components/automate-ui/src/app/entities/data-bags/data-bag-details.effects.ts:63]:
```suggestion
    map(({ payload: { id: id } }: CreateDataBagItemSuccess) => {
```

File: components/automate-ui/src/app/modules/infra-proxy/create-databag-item-modal/create-databag-item-modal.component.spec.ts (hunk 3 above)
Comment (chaitali-mane) [reply components/automate-ui/src/app/modules/infra-proxy/create-databag-item-modal/create-databag-item-modal.component.spec.ts:55]: @vivekshankar1 This is default test case added by Angular To check Component created or not. Please have a look other test cases, if I missed to add any test please let me know will add. Thank you.

Comment (vivekshankar1) [reply components/automate-ui/src/app/modules/infra-proxy/create-databag-item-modal/create-databag-item-modal.component.spec.ts:55]: Yes correct @chaitali-mane , my bad this file actually has all the required tests. Thank you.

**Review by vivekshankar1 — APPROVED** [review]
Its working fine and works as expected, thank you

**Review by shaik80 — COMMENTED** [review]
https://user-images.githubusercontent.com/47217471/111774434-78247980-88d5-11eb-8bc4-f8f3ecdc8e60.mov

can u explain to me, how can I see databag UI, to test UI changes

I run `rebuild ./components/infra-proxy-service/` in hab studio
then make serve in automate-UI also my all services are working fine,
even though I'm getting only loader
Screenshot 2021-03-19 at 5 19 59 PM

**Review by shaik80 — APPROVED** [review]

**Review by punitmundra — APPROVED** [review]

**Comment by chaitali-mane** [comment]
While acceptance testing, observed :
While creating the first Databag item I do not see the data bag item immediately.
will cover in [#4873](https://github.com/chef/automate/issues/4873)

**Pull request merged**